registry/config.go

```diff
@@ -63,6 +65,10 @@ var (
 	emptyServiceConfig = newServiceConfig(ServiceOptions{})
 )
 
+var (
+	validHostnameRegex = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
```

Comment:
I was looking this morning if [`url.Parse()`](https://golang.org/pkg/net/url/#Parse) could be used to validate, but I was unable to get it invalidate incorrect hostnames 😞